examples/structured_data/classification_with_tfdf.py

```diff
@@ -126,43 +115,66 @@
     CATEGORICAL_FEATURES_WITH_VOCABULARY.keys()
 )
 
+"""
+Now we perform basic data preparation.
+"""
+
+
+def prepare_dataframe(dataframe):
+    # Convert the target labels from string to integer.
+    dataframe[TARGET_COLUMN_NAME] = dataframe[TARGET_COLUMN_NAME].map(
+        TARGET_LABELS.index
+    )
+    # Cast the categorical features to string.
+    for feature_name in CATEGORICAL_FEATURES_WITH_VOCABULARY:
+        dataframe[feature_name] = dataframe[feature_name].astype(str)
+
+
+prepare_dataframe(train_data)
+prepare_dataframe(test_data)
+
+"""
+Now let's show the shapes of the training and test dataframes, and display some instances.
+"""
+
+print(f"Train data shape: {train_data.shape}")
+print(f"Test data shape: {test_data.shape}")
+print(train_data.head().T)
+
 """
 ## Configure hyperparameters
 
 You can find all the parameters of the Gradient Boosted Tree model in the
 [documentation](https://www.tensorflow.org/decision_forests/api_docs/python/tfdf/keras/GradientBoostedTreesModel)
 """
 
-GROWING_STRATEGY = "BEST_FIRST_GLOBAL"
-NUM_TREES = 250
-MIN_EXAMPLES = 6
-MAX_DEPTH = 5
-SUBSAMPLE = 0.65
-SAMPLING_METHOD = "RANDOM"
-VALIDATION_RATIO = 0.1
+NUM_TREES = 250  # Maximum number of decision trees. The effective number of trained trees can be smaller if early stopping is enabled.
+MIN_EXAMPLES = 6  # Minimum number of examples in a node.
+MAX_DEPTH = (
+    5  # Maximum depth of the tree. max_depth=1 means that all trees will be roots.
+)
+SUBSAMPLE = 0.65  # Ratio of the dataset (sampling without replacement) used to train individual trees for the random sampling method.
+SAMPLING_METHOD = (
+    "RANDOM"  # Control the sampling of the datasets used to train individual trees.
+)
+VALIDATION_RATIO = 0.1  # Ratio of the training dataset used to monitor the training. Require to be >0 if early stopping is enabled.
```

Comment:
In this section, move the comment descriptions to be above the variable (to keep lines short). Long descriptions should be split over 2 lines

Reply:
Done